code/modules/clothing/under/ties.dm

```diff
@@ -411,6 +411,50 @@
 	desc = "An armband, worn by the rookie nurses to display they are still not doctors. This one is dark red."
 	icon_state = "nurse"
 
+/obj/item/clothing/accessory/armband/squad
+	name = "squad armband"
+	desc = "An armband in squad colors, worn for ease of idenfication."
+	icon_state = "armband_squad"
+	var/dummy_icon_state = "armband_%SQUAD%"
+	var/static/list/valid_icon_states
+	item_state_slots = null
+
+/obj/item/clothing/accessory/armband/squad/Initialize(mapload, ...)
+	. = ..()
+	if(!valid_icon_states)
+		valid_icon_states = icon_states(icon)
+	adapt_to_squad()
+
+/obj/item/clothing/accessory/armband/squad/proc/update_clothing_wrapper(mob/living/carbon/human/wearer)
+	SIGNAL_HANDLER
+
+	var/is_worn_by_wearer = recursive_holder_check(src) == wearer
+	if(is_worn_by_wearer)
+		update_clothing_icon()
+	else
+		UnregisterSignal(wearer, COMSIG_SET_SQUAD)
+
+/obj/item/clothing/accessory/armband/squad/update_clothing_icon()
+	adapt_to_squad()
+
+/obj/item/clothing/accessory/armband/squad/pickup(mob/user, silent)
+	. = ..()
+	adapt_to_squad()
+
+/obj/item/clothing/accessory/armband/squad/equipped(mob/user, slot, silent)
+	RegisterSignal(user, COMSIG_SET_SQUAD, PROC_REF(update_clothing_wrapper), TRUE)
+	adapt_to_squad()
+	return ..()
+
+/obj/item/clothing/accessory/armband/squad/proc/adapt_to_squad()
+	var/squad_color = "squad"
+	var/mob/living/carbon/human/wearer = recursive_holder_check(src)
+	if(istype(wearer) && wearer.assigned_squad)
+		var/squad_name = lowertext(wearer.assigned_squad.name)
+		if("armband_[squad_name]" in valid_icon_states)
+			squad_color = squad_name
+	icon_state = replacetext(initial(dummy_icon_state), "%SQUAD%", squad_color)
+
```

Comment:
```suggestion
inv_overlay = image("icon" = inv_overlay_icon, "icon_state" = "[item_state? "[item_state]" : "[icon_state]"]")
if(has_suit)
	has_suit.overlays += get_inv_overlay()
	
```

Reply:
implementing these two suggestions should fix the bug.
This is because the overlay that gets attached to the suit is generated when the item is created-- and then applied and unapplied to the host suit when attached and detached-- since it's only generated once when the item is created, its' never updated as the icon is in adapt_to_squad. 

The new code removes the old overlay, regenerates it, and readds it when adapt_to_squad is called.

https://github.com/user-attachments/assets/a487bebb-9699-4bc1-9c3a-62f21976d4cc